lib/simplifyExpression/substituteScopeSearch/index.js

```diff
@@ -0,0 +1,54 @@
+const ChangeTypes = require('../../ChangeTypes');
+const Node = require('../../node');
+const TreeSearch = require('../../TreeSearch');
+
+// Searches through the tree, prioritizing deeper nodes, and substitutes
+// in-scope values for their respective expressions on a symbol node if possible.
+// Returns a Node.Status object.
+const search = TreeSearch.postOrder(scopeSubstitution);
+
+function scopeSubstitution(node, scope) {
+  if (Node.Type.isSymbol(node)) {
+    return substituteAndSimplifySymbols(node, scope);
+  }
+  else {
+    return Node.Status.noChange(node);
+  }
+}
+
+// SUBSTITUTES
+// Returns a Node.Status object with substeps
+function substituteAndSimplifySymbols(node, scope) {
+  if (!Node.Type.isSymbol(node)) {
+    return Node.Status.noChange(node);
+  }
+
+  let symbolName;
+  if (node.type === 'SymbolNode') {
```

Comment:
I left this alone for now since you followed up with the change to `allowUnaryMinus`. Let me know if we should revisit.